caPutLogApp/caPutJsonLogTask.cpp

```diff
@@ -377,6 +377,7 @@ void CaPutJsonLogTask::addPutToQueue(LOGDATA * plogData)
     flag = call; \
     if (flag != yajl_gen_status_ok) { \
         errlogSevPrintf(errlogMinor, "caPutJsonLog: JSON generation error\n"); \
+        yajl_gen_free(handle); \
```

Comment:
In c++ the conventional way to avoid this sort of resource leak would be to wrap the handle in a class with `yajl_gen_free()` in a destructor.  eg.

https://github.com/epics-base/pvDataCPP/blob/2547514abb90875bf47dda728ecaa9e30a5ab354/src/json/parseinto.cpp#L289-L300

https://github.com/epics-base/pvDataCPP/blob/2547514abb90875bf47dda728ecaa9e30a5ab354/src/json/parseinto.cpp#L329-L335

Reply:
> In c++ the conventional way to avoid this sort of resource leak would be to wrap the handle in a class with yajl_gen_free() in a destructor. eg.

That looks clean and safe. Since this code has no such wrapper class at the moment is it worth introducing it now for purpose at hand? Maybe.

If the ugliness of accessing `handle` from the macro is what we are solving, wouldn't passing it in as the first arg to the macro be acceptable?
```
#define CALL_YAJL_FUNCTION_AND_CHECK_STATUS(handle, flag, call) \
...
```

Reply:
I think that @mdavidsaver 's solution seems nicer to me; it's a pretty thin wrapper class to guarantee that we are going to fix this and other potential memory issues. Passing the handle to the macro as well feels more like a bandaid.